astroquery/mast/core.py

```diff
@@ -0,0 +1,398 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+MAST Portal
+===========
+
+Module to query the Barbara A. Mikulski Archive for Space Telescopes (MAST).
+
+"""
+
+from __future__ import print_function, division
+
+import warnings
+import json
+import time
+import os
+
+import numpy as np
+
+try: # Python 3.x
+    from urllib.parse import quote as urlencode
+except ImportError:  # Python 2.x
+    from urllib import pathname2url as urlencode
+
+from requests import HTTPError
+
+import astropy.units as u
+import astropy.coordinates as coord
+from astropy.table import Table, Row, vstack
+
+from ..query import BaseQuery
+from ..utils import commons, async_to_sync
+from ..utils.class_or_instance import class_or_instance
+from ..exceptions import TimeoutError, InvalidQueryError
+from . import conf
+
+
+__all__ = ['Observations', 'ObservationsClass',
+           'Mast',  'MastClass']
+
+
+class ResolverError(Exception):
+    pass
+
+
+def _prepare_service_request_string(jsonObj):
+    """
+    Takes a mashup json request object and turns it into a url-safe string.
+
+    Parameters
+    ----------
+    jsonObj : dict
+        A Mashup request json object (python dictionary)
+        
+    Returns
+    -------
+    response : str
+        URL encoded Mashup Request string.
+    """
+    requestString = json.dumps(jsonObj)
+    requestString = urlencode(requestString)
+    return "request="+requestString
+
+
+def _mashup_json_to_table(jsonObj):
+    """
+    Takes a json object as returned from a Mashup request and turns it into an astropy Table.
```

Comment:
I prefer `astropy` not be capitalized, but perhaps have it in double-backticks.  Astropy the organization should be capitalized, `astropy` the package, not.  However, I've never seen any official convention on this - is there one in python?  in astropy?